rerun-failed-run/README.md

```diff
@@ -1,66 +0,0 @@
-# rerun-failed-run
```

Comment:
I don't want to lose this content here, I suggest making a `README.md` in `github/workflows` and putting it there. That readme file can be used for more documentation in the future, for the other workflows in that directory.